### PR TITLE
Minor ClassMethodAssignManipulator improvements

### PR DESCRIPTION
**Pull request by staabm**

### CODE REVIEW

File: src/NodeManipulator/ClassMethodAssignManipulator.php [hunk 1]
```diff
@@ -212,21 +212,16 @@ private function collectReferenceVariableNames(ClassMethod $classMethod): array
...
         foreach ($variables as $variable) {
-            if ($this->nodeNameResolver->isName($variable, 'this')) {
+            $variableName = $this->nodeNameResolver->getName($variable);
+            if ($variableName === 'this') {
                 continue;
             }
 
             $parentNode = $variable->getAttribute(AttributeKey::PARENT_NODE);
             if ($parentNode instanceof Node && $this->isExplicitlyReferenced($parentNode)) {
-                $variableName = $this->nodeNameResolver->getName($variable);
```
Comment (staabm) [inline src/NodeManipulator/ClassMethodAssignManipulator.php:225]: removed multiple resolving of the variable name

File: src/NodeManipulator/ClassMethodAssignManipulator.php [hunk 2]
```diff
@@ -332,19 +322,19 @@ private function isFuncCallWithReferencedArgument(Node $node, Variable $variable
             return false;
         }
 
-        if (! $this->nodeNameResolver->isNames($node, ['array_shift', '*sort'])) {
+        if (! isset($node->args[0])) {
```
Comment (staabm) [inline src/NodeManipulator/ClassMethodAssignManipulator.php:326]: cheap checks first

File: src/NodeManipulator/ClassMethodAssignManipulator.php [hunk 3]
```diff
@@ -212,21 +212,16 @@ private function collectReferenceVariableNames(ClassMethod $classMethod): array
     {
         $referencedVariables = [];
 
-        /** @var Variable[] $variables */
```
Comment (staabm) [inline src/NodeManipulator/ClassMethodAssignManipulator.php]: because of the generics on the finder, we don't need the inline `@vars`

**Comment by TomasVotruba** [comment]
Thanks :+1:  

Could you share some perf change numbers too? So we see how this helps.

**Comment by staabm** [comment]
> Could you share some perf change numbers too

in this PR the perf benefit is pretty minimal.

**Comment by staabm** [comment]
should be good to go

**Comment by TomasVotruba** [comment]
Let's ship it then, thanks :+1:

**Pull request merged**